server/src/infra/entities/file-history.entity.ts

```diff
@@ -0,0 +1,34 @@
+import { Column, DeleteDateColumn, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+import { AssetEntity } from './asset.entity';
+
+@Entity('file_history')
+export class FileHistoryEntity {
+  @PrimaryGeneratedColumn('uuid')
+  id!: string;
+
+  @ManyToOne(() => AssetEntity, { onDelete: 'CASCADE', onUpdate: 'CASCADE' })
+  asset!: AssetEntity;
+
+  @Column({ type: 'varchar' })
+  assetId!: string;
+
+  @Column({ type: 'varchar' })
+  pathType!: PathType;
```

Comment:
Is it possible to drop this column altogether? A generic string isn't great for type safety and I don't see it used for anything.

Reply:
`pathType` will later be necessary because it enables us to find a "lost" file that hadn't been persisted to the asset database correctly. The combination `entityId` + `pathType` should always be unique (among the none soft-deleted move entities).